android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginScreen.kt

```diff
@@ -71,81 +132,140 @@ fun LoginScreenInner(
                     modifier = Modifier.size(280.dp),
                 )
             }
-            Row(modifier = Modifier.weight(1f).fillMaxSize()) {
-                LoginForm(email, password, onEmailChange, onPasswordChange, onLoginPress)
-            }
+            Row(modifier = Modifier.weight(1f).fillMaxSize()) { LoginForm(state, events, 280.dp) }
         }
     }
 }
 
 @Composable
-fun LoginForm(
-    email: String,
-    password: String,
-    onEmailChange: (String) -> Unit,
-    onPasswordChange: (String) -> Unit,
-    onLoginPress: () -> Unit,
+private fun LoginForm(
+    state: LoginState,
+    events: LoginEvents,
+    width: Dp,
     modifier: Modifier = Modifier,
 ) {
     Column(
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.spacedBy(4.dp),
         modifier = modifier.fillMaxSize(),
     ) {
-        EmailField(email, onEmailChange)
-        PasswordField(password, onPasswordChange)
-        Row(modifier = Modifier.width(280.dp), horizontalArrangement = Arrangement.spacedBy(4.dp)) {
-            RegisterButton(modifier = Modifier.weight(1f)) {}
-            LoginButton(modifier = Modifier.weight(1f), onLoginPress)
+        LoginFields(state, events, Modifier.width(width))
+        AnimatedVisibility(!state.registering) {
+            ResetText(
+                stringResource(R.string.forgot_password),
+                events.onResetPress,
+                Modifier.width(width),
+            )
         }
+        LoginButton(
+            stringResource(if (state.registering) R.string.register else R.string.login),
+            if (state.registering) events.onRegisterPress else events.onLoginPress,
+            Modifier.width(width),
+        )
+        SwitchText(
+            stringResource(
+                if (state.registering) R.string.already_registered else R.string.not_registered
+            ),
+            stringResource(
+                if (state.registering) R.string.login_instead else R.string.register_instead
+            ),
+            events.onSwitchPress,
+            Modifier.width(width),
+        )
     }
 }
 
 @Composable
-fun EmailField(text: String, onValueChange: (String) -> Unit, modifier: Modifier = Modifier) {
-    OutlinedTextField(
-        value = text,
-        onValueChange = { onValueChange(it) },
-        label = { Text(stringResource(R.string.email)) },
-        leadingIcon = {
-            Icon(
-                imageVector = ImageVector.vectorResource(R.drawable.email),
-                contentDescription = stringResource(R.string.email_icon),
+private fun LoginFields(state: LoginState, events: LoginEvents, modifier: Modifier = Modifier) {
+    Column(modifier) {
+        AnimatedVisibility(state.registering) {
+            LoginField(
```

Comment:
Although the API technically allow empty name, I think we should sanitize it at in this form. So would be nice If we can check if the field is empty here